returnn/tf/layers/basic.py

```diff
@@ -1341,6 +1341,17 @@ def __init__(self, position, axis, **kwargs):
     # (BatchAxes.., InputAxesBeforeGatherAxis, PositionAxes.., InputAxesAfterGatherAxis..)
     self.output.placeholder = tf.gather(params=params, indices=indices, axis=gather_axis, batch_dims=batch_dims)
 
+    if input_data.dim_tags[old_gather_axis].is_batch_dim():
+      for axis in self.output.size_placeholder:
```

Comment:
Instead of iterating over the `size_placeholder`, you should iterate over the dim tags.